### PR TITLE
Update to Avalonia 11 RC 1

### PR DESCRIPTION
**Pull request by Numpsy**
Just another set of test changes to build with the latest Avalonia 11 nightly, as you can never break too many things

### CODE REVIEW

File: src/Avalonia.FuncUI/DSL/Base/Animatable.fs [hunk 1]
```diff
@@ -10,5 +10,5 @@ module Animatable =
         static member transitions<'t when 't :> Animatable>(transitions: Transitions) : IAttr<'t> =
             AttrBuilder<'t>.CreateProperty<Transitions>(Animatable.TransitionsProperty, transitions, ValueNone)
             
-        static member clock<'t when 't :> Animatable>(clock: IClock) : IAttr<'t> =
-            AttrBuilder<'t>.CreateProperty<IClock>(Animatable.ClockProperty, clock, ValueNone)
\ No newline at end of file
+        // static member clock<'t when 't :> Animatable>(clock: IClock) : IAttr<'t> =
```
Comment (Numpsy) [inline src/Avalonia.FuncUI/DSL/Base/Animatable.fs:13]: refs https://github.com/AvaloniaUI/Avalonia/pull/11552

File: src/Avalonia.FuncUI/DSL/MenuBase.fs [hunk 2]
```diff
@@ -10,7 +10,7 @@ module MenuBase =
     type MenuBase with
             
         static member onMenuOpened<'t when 't :> MenuBase>(func: RoutedEventArgs -> unit, ?subPatchOptions) : IAttr<'t> =
-            AttrBuilder<'t>.CreateSubscription(MenuBase.MenuOpenedEvent, func, ?subPatchOptions = subPatchOptions);
+            AttrBuilder<'t>.CreateSubscription(MenuBase.OpenedEvent, func, ?subPatchOptions = subPatchOptions);
```
Comment (Numpsy) [inline src/Avalonia.FuncUI/DSL/MenuBase.fs:13]: refs https://github.com/AvaloniaUI/Avalonia/pull/11549

File: src/Avalonia.FuncUI/DSL/TabItem.fs [hunk 3]
```diff
@@ -5,14 +5,15 @@ module TabItem =
...
+    open System
+
     let create (attrs: IAttr<TabItem> list): IView<TabItem> =
         ViewBuilder.Create<TabItem>(attrs)
      
     type TabItem with
 
-        static member tabStripPlacement<'t when 't :> TabItem>(placement: Dock) : IAttr<'t> =
-            AttrBuilder<'t>.CreateProperty<Dock>(TabItem.TabStripPlacementProperty, placement, ValueNone)
+        static member tabStripPlacement<'t when 't :> TabItem>(placement: Dock Nullable) : IAttr<'t> =
```
Comment (Numpsy) [inline src/Avalonia.FuncUI/DSL/TabItem.fs:15]: refs https://github.com/AvaloniaUI/Avalonia/pull/11366

File: src/Avalonia.FuncUI/DSL/TransitioningContentControl.fs [hunk 4]
```diff
@@ -13,8 +13,8 @@ module TransitioningContentControl =
 
     type TransitioningContentControl with
 
-        static member currentContent<'t when 't :> TransitioningContentControl>(value: obj) : IAttr<'t> =
-            AttrBuilder<'t>.CreateProperty<obj>(TransitioningContentControl.CurrentContentProperty, value, ValueNone)
+        //static member currentContent<'t when 't :> TransitioningContentControl>(value: obj) : IAttr<'t> =
```
Comment (Numpsy) [inline src/Avalonia.FuncUI/DSL/TransitioningContentControl.fs:16]: refs https://github.com/AvaloniaUI/Avalonia/pull/11326

File: src/Avalonia.FuncUI/VirtualDom/VirtualDom.fs [hunk 5]
```diff
@@ -16,7 +16,7 @@ module rec VirtualDom =
         let delta = Differ.diff(last, next)
         Patcher.patch(root, delta)
 
-    let updateRoot (host: IContentControl, last: IView option, next: IView option) =
+    let updateRoot (host: ContentControl, last: IView option, next: IView option) =
         let root : Control voption =
```
Comment (Numpsy) [inline src/Avalonia.FuncUI/VirtualDom/VirtualDom.fs:20]: refs https://github.com/AvaloniaUI/Avalonia/pull/11477

File: src/Examples/Component Examples/Examples.Presso/Lib/Library.fs [hunk 6]
```diff
@@ -18,5 +18,4 @@ module Extensions =
             if uri.IsAbsoluteUri && uri.IsFile then
                 new Bitmap(uri.LocalPath)
             else
-                let assets = AvaloniaLocator.Current.GetService<IAssetLoader>();
-                new Bitmap(assets.Open(uri));
+                new Bitmap(AssetLoader.Open(uri))
```
Comment (Numpsy) [inline src/Examples/Component Examples/Examples.Presso/Lib/Library.fs:21]: refs https://github.com/AvaloniaUI/Avalonia/pull/11190

**Comment by Numpsy** [comment]
Ok, Avalonia 11 now has an actual release candidate package available :-)

**Comment by Numpsy** [comment]
So it's building ok, and the control catalog seems to work :-)

There are a bunch of new warnings about some style related things being deprecated, that I haven't looked at, e.g.
```
/home/runner/work/Avalonia.FuncUI/Avalonia.FuncUI/src/Avalonia.FuncUI/Components/StaticComponent.fs(24,15): warning FS0044: This construct is deprecated. This interface may be removed in 12.0. Use StyledElement, or override StyledElement.StyleKeyOverride to override the StyleKey for a class
```

File: src/Avalonia.FuncUI/DSL/Base/Animatable.fs (hunk 1 above)
Comment (Numpsy) [reply src/Avalonia.FuncUI/DSL/Base/Animatable.fs:13]: Ok to just remove these commented out bits?

**Review by sleepyfran — APPROVED** [review]
Amazing job! We can definitely merge with the warnings, since the deprecation notice is for v12 we can either attempt to address it in a separate PR before the final v11 release (if any of us have any time to do so) or just fix all of them once they're removed in v12 :^)

**Comment by JaggerJo** [comment]
I'm merging this.

**Pull request merged**